src/full/Agda/Compiler/Malfunction/Compiler.hs

```diff
@@ -0,0 +1,627 @@
+  TSort             -> error ("Unimplemented " ++ show tt)
+  TErased           -> return wildcardTerm -- TODO: so... anything can go here?
+  TError TUnreachable -> return wildcardTerm
+
+-- | We use this when we don't care about the translation.
+wildcardTerm :: Term
+wildcardTerm = nullary $ errorT "__UNREACHABLE__"
+
+nullary :: Term -> Term
+nullary = Mlambda []
+
+indexToVarTerm :: MonadReader Env m => Int -> m Term
+indexToVarTerm i = do
+  ni <- asks _level
+  return (Mvar (ident (ni - i - 1)))
+
+-- translateSwitch :: MonadReader m => Term -> TAlt -> m ([Case], Term)
+-- translateSwitch tcase alt = case alt of
+-- --  TAGuard c t -> liftM2 (,) (pure <$> translateCase c) (translateTerm t)
+--   TALit pat body -> do
+--     b <- translateTerm body
+--     let c = pure $ litToCase pat
+--     return (c, b)
+--   TACon con arity t -> do
+--     tg <- nameToTag con
+--     usedFields <- snd <$> introVars arity
+--       (Set.map (\ix -> arity - ix - 1) . Set.filter (<arity) <$> usedVars t)
+--     (vars, t') <- introVars arity (translateTerm t)
+--     let bt = bindFields vars usedFields tcase t'
+--           -- TODO: It is not clear how to deal with bindings in a pattern
+--     return (pure tg, bt)
+--   TAGuard gd rhs -> return ([], Mvar "TAGuard.undefined")
+
+translateAltsChain :: MonadReader Env m => Term -> Maybe Term -> [TAlt] -> m [([Case], Term)]
+translateAltsChain tcase defaultt [] = return $ maybe [] (\d -> [(defaultCase, d)]) defaultt
+translateAltsChain tcase defaultt (ta:tas) =
+  case ta of
+    TALit pat body -> do
+      b <- translateTerm body
+      let c = litToCase pat
+      (([c], b):) <$> go
+    TACon con arity t -> do
+      tg <- nameToTag con
+      usedFields <- snd <$> introVars arity
+        (Set.map (\ix -> arity - ix - 1) . Set.filter (<arity) <$> usedVars t)
+      (vars, t') <- introVars arity (translateTerm t)
+      let bt = bindFields vars usedFields tcase t'
+      -- TODO: It is not clear how to deal with bindings in a pattern
+      (([tg], bt):) <$> go
+    TAGuard grd t -> do
+      tgrd <- translateTerm grd
+      t' <- translateTerm t
+      tailAlts <- go
+      return [(defaultCase,
+                Mswitch tgrd
+                [(trueCase, t')
+                , (defaultCase, Mswitch tcase tailAlts)])]
+  where
+    go = translateAltsChain tcase defaultt tas
+
+defaultCase :: [Case]
+defaultCase = [CaseAnyInt, Deftag]
+
+bindFields :: [Ident] -> Set Int -> Term -> Term -> Term
+bindFields vars used termc body = case map bind varsRev of
+  [] -> body
+  binds -> Mlet binds body
+  where
+    varsRev = zip [0..] (reverse vars)
+    arity = length vars
+    bind (ix, iden)
+      -- TODO: we bind all fields. The detection of used fields is bugged.
+      | True || Set.member ix used = Named iden (Mfield ix termc)
+      | otherwise = Named iden wildcardTerm
+
+litToCase :: Literal -> Case
+litToCase l = case l of
+  LitNat _ i -> CaseInt . fromInteger $ i
+  _          -> error "Unimplemented"
```

Comment:
If I saw correctly, malformed does not support e.g. string literal alternatives? You can use https://github.com/agda/agda/blob/master/src/full/Agda/Compiler/Treeless/EliminateLiteralPatterns.hs to convert literal alternatives to if-else cascades. (Maybe they should be converted to guard alternatives instead)